packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmTypeFactory.cs

```diff
@@ -98,8 +98,33 @@ private ExtensibleEnumSerializationProvider[] CreateExtensibleEnumSerializations
                 }
             }
 
-            ClientCache[inputClient] = client;
-            return client;
+            var result = client is not null && IsValidClient(inputClient, client) ? client : null;
+            ClientCache[inputClient] = result;
+            return result;
+        }
+
+        private bool IsValidClient(InputClient inputClient, ClientProvider client)
+        {
+            // client is not valid if it has no operations, sub-client methods, or custom code methods
+            if (inputClient.Operations.Count > 0)
+            {
+                return true;
+            }
+
+            foreach (var subclient in client.SubClients)
+            {
+                if (subclient.Methods.Count > 0)
+                {
+                    return true;
```

Comment:
We need to also check to see if the subclients have custom code defined.

Reply:
checked but only check methods of custom code for now.